src/apps/competitions/tasks.py

```diff
@@ -384,16 +366,17 @@ def unpack_competition(competition_dataset_pk):
 
                     if 'key' in solution:
                         # add {index: {'key': key, 'tasks': task_index}} to competition solutions
-                        # TODO:// raise an exception if solution matching key doesn't exist
-                        Solution.objects.get(key=solution['key']).tasks.add(*Task.objects.filter(key__in=task_keys))
+                        solution = Solution.objects.filter(key=solution['key']).first()
+                        if not solution:
+                            raise CompetitionUnpackingException(f'Could not find solution with key: "{solution["key"]}"')
+                        solution.tasks.add(*Task.objects.filter(key__in=task_keys))
 
                         competition['solutions'][index] = solution['key']
 
                     else:
                         # create solution object and then add {index: {'key': key, 'tasks': task_indexes}} to competition solutions
-                        name = solution[
-                            'name'] if 'name' in solution else f"solution @ {now().strftime('%m-%d-%Y %H:%M')}"
-                        description = solution['description'] if 'description' in solution else None
+                        name = solution.get('name') or f"solution @ {now().strftime('%m-%d-%Y %H:%M')}"
```

Comment:
Can do f string formatting of dates like:

```python
f"solution @ {now():%m-%d-%Y %H:%M}"
```

Reply:
brilliant!

Reply:
Python is so pretty :)